bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/DataLayerConfig.java

```diff
@@ -39,4 +41,29 @@
     @Property(label="Data Layer client library not included")
     boolean skipClientlibInclude() default false;
 
+    /**
+     *
+     * @return the name of the datalayer
+     * Defaults to adobeDataLayer.
+     */
+    @Property(label = "Data Layer object name")
+    String name() default DATALAYER_OBJECT_NAME_ADOBE;
+
+    /**
+     *
+     * @return the type of push function (push() or gtag()) of the datalayer
+     * Defaults to push.
+     */
+    @Property(label = "Use gtag function to push data")
```

Comment:
Wouldn't it be enough to know that the name of the datalayer is gtag?

Reply:
no, the datalayer can be called anything or normally dataLayer, regardless of whether GA (gtag) or GTM is used.